src/pgas.jl

```diff
@@ -0,0 +1,151 @@
+""" 
+    initialization(model::AbstractStateSpaceModel)
+
+Define the distribution of the initial state of the State Space Model
+"""
+function initialization end
+
+""" 
+    transition(model::AbstractStateSpaceModel, state, step)
+
+Define the transition density of the State Space Model
+Must return `nothing` if it consumed all the data
+"""
+function transition end
+
+"""
+    observation(model::AbstractStateSpaceModel, state, step)
```

Comment:
Are the variables `state` and `step` redundant with each other, i.e. `state = model.X[step]`? If so, maybe we can remove the extra variable `state` here? Otherwise, perhaps add a comment explaining why. 

EDIT: it seems this is requires in [line](https://github.com/TuringLang/AdvancedPS.jl/pull/32/files#diff-4e0ce155dd587f5f2009d7664a4e1db765a58990171122250646bf9a827aa3c3R102). Let's add a comment explain why it is needed.